CONTRIBUTING.md

```diff
@@ -47,73 +65,82 @@ Instead, do something like this:
-;;       React Element directly.
+;;       React element directly.
 ```
 
-In JavaScript, you can do things like:
+In non-React JavaScript, you can do things like:
+
 ```javascript
 var myDiv = document.createElement('div', ...);
-myDiv.focus()
+myDiv.focus();
 ```
 or
+
 ```javascript
 var myDiv = document.createElement('div', ...);
-SomeThirdPartyLibraryThatTakesADomNode(myDiv)
+SomeThirdPartyLibraryThatTakesADomNode(myDiv);
 ```
 
-React Elements may not be substituted for DOM nodes in these cases. You must use a `ref` (see React's documentation) to obtain access to the DOM node once React has rendered it into the browser window.
+In situations where a method operates on a DOM node, React elements may not be substituted. You must use a `ref` ([see React's documentation](https://facebook.github.io/react/docs/refs-and-the-dom.html)) to obtain access to the DOM node once React has rendered it into the browser window.
 
-### State Set -> Read is Unreliable
+### Set state → read state: It doesn't work the way you think it should
 
-[State updates may be asynchronous](https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous), meaning that `state` will not contain new values when set, but instead will have those values after the re-render. For example:
+[State updates are not immediate](https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous), meaning that `state` will not immediately contain a new value after you set it, but instead will have that value after the next re-render. For example:
 
 ```clojure
 (swap! state assoc :foo 17)
-(get @state :foo) ; <- :foo is not 17 here!
+(get @state :foo) ; <- :foo has yet to be changed to 17 here!
 ```
 
-So, instead of:
+So, instead of immediately reading a value back from state:
+
 ```clojure
 (swap! state assoc :foo (bar ...))
 (some-func (:foo @state))
 ```
 
-try:
+use the new value directly:
+
 ```clojure
 (let [new-value (bar ...)]
   (swap! state assoc :foo new-value)
   (some-func new-value))
 ```
 
-Instead of:
-```clojure
-(react/call :some-state-modifying-method this)
-(some-func (:some-key @state))
-```
+or wait until after the re-render:
 
-try:
 ```clojure
 (react/call :some-state-modifying-method this)
 (after-update #(some-func (:some-key @state))))
 ```
 
-### Avoid Infinite Loops
+### Avoid infinite loops
+
+Changing state causes a re-render. If you update state in a lifecycle method, this can lead to a loop:
 
-Changing state causes a re-render. If you change state in a lifecycle method such as `component-did-update`, this will re-render, which will call `component-did-update` again, resulting in a loop.
+1. state changes in `component-did-update`
+2. state change starts re-render
+3. re-render calls `component-did-update`
+4. state changes in `component-did-update`
+5. state change starts re-render
+6. ...
+
+So, avoid changing states inside of lifecycle methods.
```

Comment:
`component-will-receive-props` is a lifecycle method, but changing state there would be a very reasonable thing to do.